libnavigation-core/src/test/java/com/mapbox/navigation/core/routealternatives/AlternativeRouteProgressDataProviderTest.kt

```diff
@@ -0,0 +1,289 @@
+        val alternativeMetadata = alternativeRouteMetadata(
+            primaryForkLegIndex = 5,
+            primaryForkRouteGeometryIndex = primaryRouteGeometryIndex - 1,
+            primaryForkLegGeometryIndex = 80,
+            alternativeForkLegIndex = 3,
+            alternativeForkRouteGeometryIndex = 90,
+            alternativeForkLegGeometryIndex = 40,
+            emptyList()
+        )
+        val expected = RouteProgressData(3, 90, 40)
+
+        val actual = AlternativeRouteProgressDataProvider.getRouteProgressData(
+            primaryRouteProgressData,
+            alternativeMetadata
+        )
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `fork passed, alternative is longer than primary`() = coroutineRule.runBlockingTest {
+        val primaryRouteGeometryIndex = 200
+        val primaryRouteProgressData = RouteProgressData(3, primaryRouteGeometryIndex, 40)
+        val alternativeMetadata = alternativeRouteMetadata(
+            primaryForkLegIndex = 3,
+            primaryForkRouteGeometryIndex = primaryRouteGeometryIndex - 1,
+            primaryForkLegGeometryIndex = 40,
+            alternativeForkLegIndex = 5,
+            alternativeForkRouteGeometryIndex = 130,
+            alternativeForkLegGeometryIndex = 80,
+            emptyList()
+        )
+        val expected = RouteProgressData(5, 130, 80)
+
+        val actual = AlternativeRouteProgressDataProvider.getRouteProgressData(
+            primaryRouteProgressData,
+            alternativeMetadata
+        )
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `on fork`() = coroutineRule.runBlockingTest {
+        val primaryRouteGeometryIndex = 200
+        val primaryRouteProgressData = RouteProgressData(4, primaryRouteGeometryIndex, 30)
+        val alternativeMetadata = alternativeRouteMetadata(
+            primaryForkLegIndex = 5,
+            primaryForkRouteGeometryIndex = primaryRouteGeometryIndex,
+            primaryForkLegGeometryIndex = 80,
+            alternativeForkLegIndex = 3,
+            alternativeForkRouteGeometryIndex = 90,
+            alternativeForkLegGeometryIndex = 40,
+            emptyList()
+        )
+        val expected = RouteProgressData(3, 90, 40)
+
+        val actual = AlternativeRouteProgressDataProvider.getRouteProgressData(
+            primaryRouteProgressData,
+            alternativeMetadata
+        )
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `before fork, alternative starts together with the primary route`() = coroutineRule.runBlockingTest {
+        val primaryRouteProgressData = RouteProgressData(4, 200, 30)
+        val alternativeMetadata = alternativeRouteMetadata(
+            primaryForkLegIndex = 5,
+            primaryForkRouteGeometryIndex = 250,
+            primaryForkLegGeometryIndex = 80,
+            alternativeForkLegIndex = 5,
+            alternativeForkRouteGeometryIndex = 250,
+            alternativeForkLegGeometryIndex = 70,
```

Comment:
This test scenario is in theory impossible. If the fork is on the upcoming leg, the `ForkLegGeometryIndex` for both primary and alternative route should be equal. Of course there can be edge cases we mentioned above (like incidents or congestion adding geometry points) but let's clarify that in the test description. The same applies to some of the other tests. This can also be addressed separately, not a blocker.